packages/next/src/server/base-server.ts

```diff
@@ -3176,7 +3220,7 @@ export default abstract class Server<
     } else if (cachedData.kind === CachedRouteKind.APP_PAGE) {
       // If the request has a postponed state and it's a resume request we
       // should error.
-      if (cachedData.postponed && minimalPostponed) {
+      if (didPostpone && minimalPostponed) {
```

Comment:
This is mostly a note to our future selves. The fact that we repeatedly check whether we posptoned tells me this function needs to be split into two paths most likely